### PR TITLE
New package: metamath-0.179

### PR DESCRIPTION
**Pull request by zhengqunkoo**
Makefile.am is edited as per https://github.com/metamath/metamath-exe/issues/3#issuecomment-536509055

### CODE REVIEW

File: srcpkgs/metamath/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+pkgname=metamath
+version=0.179
+revision=1
+build_style=gnu-configure
+hostmakedepends="autoconf automake"
+short_desc="Constructs mathematics from scratch, starting from ZFC"
+maintainer="Zhengqun Koo <root@zhengqunkoo.com>"
+license="GPL-2.0-or-later"
+homepage="http://us.metamath.org/mpeuni/mmset.html"
+distfiles="https://github.com/metamath/metamath-exe/archive/master.tar.gz"
```
Comment (Chocimier) [inline srcpkgs/metamath/template]: We need constant address, best if tagged release.

**Comment by Chocimier** [comment]
@xelxebar, comments?

File: srcpkgs/metamath/template (hunk 1 above)
Comment (zhengqunkoo) [reply srcpkgs/metamath/template]: Found link to untagged release on primary mirror, amended commit.

Comment (Chocimier) [reply srcpkgs/metamath/template]: This is not constant either, see earlier #15721

Comment (zhengqunkoo) [reply srcpkgs/metamath/template]: Right, I didn't search before sending a pull request, sorry for the duplication.

**Comment by xtraeme** [comment]
Hmmm

**Comment by xelxebar** [comment]
Sorry for the hiatus.
It took some time, due to mailing list issues, but now upstream is working with me to put together a more Linux friendly package.

In particular, we now have a static url pointing to the source for *just* the metamath executable. There is a slight issue with the autotools scripts, but once upstream patches those, I will try to put this package together.

Since the metamath databases are by their nature non-versionable, I think it makes the most sense to have users clone the metamath/set.mm repo themselves.

Personally, I also have a wrapper shell function to help select, list, update, *etc.* the databases while also calling metamath with rlwap and initial arguments to make it feel more like a normal cli application. Do you think it would be worthwile delivering something similar in this package as well?

**Comment by zhengqunkoo** [comment]
@xelxebar I haven't had the opportunity to use metamath yet, so you're more qualified as a user to maintain this package. Feel free to open another PR.

That said, I feel that wrappers should be contributed upstream instead of here, but the final decision is of course up to the void-linux team.

**Comment by xelxebar** [comment]
@zhengqunkoo I am not entirely sure where the line falls regarding a wrapper script. Arguably, it's a distribution-specific choice, since currently most users seem to be using the everything-in-a-single-directory upstream releases. Either way, I did share my wrapper with upstream and am awaiting comments.

I am also still waiting for a reply from upstream regarding the `Makefile.am` patch; however, there is something to be said for just making the package available now instead of waiting.

If I Void agrees it's a good idea, I will go ahead and scrounge up a PR with the necessary pieces.

[0]:http://us.metamath.org/downloads/metamath-program.zip

**Comment by xelxebar** [comment]
It seems like upstream has pretty much settled on what they want to provide for the time being. I went ahead and put together pull request #18584. It packages up the exetuable without the databases. I also provide a wrapper script that helps download and use the databases.

**Comment by ericonr** [comment]
Closing in favor of #18584, since we don't want to depend on unversioned distfiles...

**Pull request closed without merging**